### ISSUE TITLE
Refactor: Use `__future__.annotations` 

### ISSUE THREAD

**Issue by provinzkraut** [starlite-api/starlite#1163]
Use `from __future__ import annotations` for all type annotations needs. 

Process can be partially automated using https://github.com/tyleryep/flake8-future-annotations to check and https://github.com/asottile/pyupgrade to rewrite annotations.

`flake8-future-annotations` should also be added to pre-commit in this process.

**Comment by peterschutt** [comment]
If we add `from __future__ import annotations` to a file, pyupgrade will update the type annotation syntax to not use the generic collection types from typing, e.g, will swap `List[...]` to `list[...]`.

This breaks pydantic on 3.8 where it requires the typing collection types for annotations.

**Comment by peterschutt** [comment]
So we could do [this workaround](https://github.com/asottile/pyupgrade/issues/622#issuecomment-1111135941):

```py
_Shape2D: TypeAlias = "_HasShape2D | Tuple[int, int] | Fraction"
```
For any internal pydantic model that has a collection attribute.

**Comment by peterschutt** [comment]
This did stop pyupgrade from overwriting the pydantic model types, but had the flow on effect that pydantic wasn't fully resolving the types. For example in the dto factory, we use the plugin to create a pydantic model type from the source model type, then we take the `__fields__` from that generated model, manipulate them and produce the dto model type. The dto model type would then have forward ref types.

For example, using a pydantic model created from this dataclass (note the type alias) via the dataclass plugin:

```py
ListInt: TypeAlias = "List[int]"

@dataclass
class DC:
    a: float = 1.01
    b: str = "string"
    c: ListInt = field(default_factory=lambda: [0, 1, 2, 3])

DTO = DataclassFactory[DC]
DTO.parse_obj({})
```
results in 

> pydantic.errors.ConfigError: field "c" not yet prepared so type is still a ForwardRef, you might need to call DC.update_forward_refs().

So, I added `localns=...` to the signature of `PluginProtocol.to_pydantic_model_class()` so that local namespace of the model can be used to manually evaluate the forward refs before creating the model.. which... I think is a good thing to do anyway? Or, at least can't hurt?

I think this gives us a path forward with this:

1. Look at all internal pydantic models and swap out their collection type annotations with aliased ones, as above.
2. Deal with any forward ref resolution issues.
3. Do the flake8/pyupgrade trick.

**Comment by provinzkraut** [comment]
Implemented in #1214 and #1216.
